src/main/java/com/sobytylnik/SQLProfileRepository.java

```diff
@@ -0,0 +1,57 @@
+package com.sobytylnik;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Resource;
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+
+@Service("profileService")
+@Transactional
+public class SQLProfileRepository implements ProfileRepository {
+
+    @Resource(name="sessionFactory")
+    private SessionFactory sessionFactory;

```

Comment:
Try to use `EntityManager` and other JPA interfaces instead of native hibernate classes.